lib/core/src/display/rendering/scene.rs

```diff
@@ -1,24 +1,99 @@
-use crate::system::web::{get_element_by_id, dyn_into, Result};
+use wasm_bindgen::prelude::Closure;
+use crate::prelude::*;
+use crate::system::web::get_element_by_id;
+use crate::system::web::dyn_into;
+use crate::system::web::Result;
+use crate::system::web::StyleSetter;
+use crate::system::web::resize_observer::ResizeObserver;
 use web_sys::HtmlElement;
 use nalgebra::Vector2;
+use std::cell::RefCell;
+use std::rc::Rc;
+use std::fmt;
+
+// ======================
+// === ResizeCallback ===
+// ======================
+
+pub type ResizeCallback = Box<dyn Fn(&Vector2<f32>)>;
+
+// =================
+// === SceneData ===
+// =================
+
+#[derive(Derivative)]
+#[derivative(Debug)]
+struct SceneData {
+    dimensions : Vector2<f32>,
+    #[derivative(Debug="ignore")]
+    callbacks : Vec<ResizeCallback>
+}
+
+impl SceneData {
+    pub fn new(dimensions : Vector2<f32>) -> Self {
+        let callbacks = default();
+        Self { dimensions, callbacks }
+    }
+}
+
+// =============
+// === Scene ===
+// =============
 
 /// A collection for holding 3D `Object`s.
-#[derive(Debug)]
+//#[derive(Debug)]
 pub struct Scene {
```

Comment:
This struct keeps scene of html objects only, or of any kind of objects?

Reply:
It's supposed to hold other objects too. I am extending it now that I am implementing a PhysicsRenderer